src/engine/srv.c

```diff
@@ -800,17 +800,20 @@ dss_xstreams_fini(bool force)
 	bool			 started = false;
 
 	D_DEBUG(DB_TRACE, "Stopping execution streams\n");
-	dss_xstreams_open_barrier();
-	rc = bio_nvme_ctl(BIO_CTL_NOTIFY_STARTED, &started);
-	D_ASSERT(rc == 0);
 
+	ABT_mutex_lock(xstream_data.xd_mutex);
 	/* Notify all xstreams to reject new ULT creation first */
 	for (i = 0; i < xstream_data.xd_xs_nr; i++) {
 		dx = xstream_data.xd_xs_ptrs[i];
 		if (dx == NULL)
 			continue;
 		ABT_future_set(dx->dx_stopping, dx);
 	}
+	ABT_cond_broadcast(xstream_data.xd_ult_barrier);
```

Comment:
why signal xd_ult_barrier here?

Reply:
Thank you! This was added by a mistake. I am going to fix it.